src/ast/seq_decl_plugin.cpp

```diff
@@ -1637,67 +1637,142 @@ seq_util::rex::info seq_util::rex::get_info_rec(expr* e) const {
             return get_info_rec(e->get_arg(0));
+        case OP_RE_PLUS:
+            i1 = get_info_rec(e->get_arg(0));
+            //r+ is not normalized if r is nullable, the normalized form would be r*
+            return info(i1.classical, i1.classical, i1.interpreted, i1.nonbranching, (i1.nullable == l_false ? i1.normalized : false), i1.monadic, false, l_true, 0, i1.star_height + 1);
         case OP_RE_COMPLEMENT:
-            return info(0); 
+            i1 = get_info_rec(e->get_arg(0));
+            nullable = (i1.nullable == l_true ? l_false : (i1.nullable == l_false ? l_true : l_undef));
+            min_length = (nullable == l_false ? 1 : 0);
+            return info(false, i1.standard, i1.interpreted, i1.nonbranching, i1.normalized, i1.monadic, false, nullable, min_length, i1.star_height);
         case OP_RE_LOOP:
             i1 = get_info_rec(e->get_arg(0));
-            return info(u.max_mul(i1.min_length, e->get_decl()->get_parameter(0).get_int()));
+            lower_bound = e->get_decl()->get_parameter(0).get_int();
+            //r{l,m} is not normalized if r is nullable but l > 0
+            normalized = (i1.nullable == l_false && lower_bound > 0 ? false : i1.normalized);
+            nullable = (i1.nullable == l_true || lower_bound == 0 ? l_true : i1.nullable);
+            return info(i1.classical, i1.classical, i1.interpreted, i1.nonbranching, normalized, i1.singleton, false, nullable, u.max_mul(i1.min_length, lower_bound), i1.star_height);
         case OP_RE_DIFF:
             i1 = get_info_rec(e->get_arg(0));
             i2 = get_info_rec(e->get_arg(1));
-            return info(std::max(i1.min_length, i2.min_length > 0 ? 0 : UINT_MAX));
+            return info(false,
+                i1.standard & i2.standard,
+                i1.interpreted & i2.interpreted,
+                i1.nonbranching & i2.nonbranching,
+                i1.normalized & i2.normalized,
+                i1.monadic & i2.monadic,
+                false,
+                ((i1.nullable == l_true && i2.nullable == l_false) ? l_true : ((i1.nullable == l_false || i2.nullable == l_false) ? l_false : l_undef)),
+                std::max(i1.min_length, i2.min_length),
+                std::max(i1.star_height, i2.star_height));
         }
         return invalid_info;
     }
     expr* c, * t, * f;
     if (u.m.is_ite(e, c, t, f)) {
         i1 = get_info_rec(t);
         i2 = get_info_rec(f);
-        return info(std::min(i1.min_length, i2.min_length));
+        min_length = std::min(i1.min_length, i2.min_length);
+        nullable = (i1.nullable == i2.nullable ? i1.nullable : l_undef);
+        //TBD: whether ite is interpreted or not depends on whether the condition is interpreted and both branches are interpreted
+        return info(false, false, false, false, i1.normalized && i2.normalized, i1.monadic && i2.monadic, i1.singleton && i2.singleton, nullable, min_length, std::max(i1.star_height, i2.star_height));
     }
     return invalid_info;
```

Comment:
Since it is possible to return invalid_info, the combiners need to be defined if one of the infos is invalid.
Currently, initialization of invalid infos leave a lot of fields uninitialized. This leads to undefined behavior.